### PR TITLE
initial plumbing for prompting on unsigned on untrusted pacakges

### PR DESCRIPTION
**Pull request by tomerd**
motivation: allow for promp UX when desired

changes: define delegate methods to allow clients to drive prompt behavior

### CODE REVIEW

**Comment by tomerd** [comment]
@swift-ci smoke test

**Comment by tomerd** [comment]
@yim-lee

File: Sources/PackageRegistry/SignatureValidation.swift [hunk 1]
```diff
@@ -203,28 +238,7 @@ struct SignatureValidation {
                 case .certificateInvalid(let reason):
                     completion(.failure(RegistryError.invalidSigningCertificate(reason: reason)))
                 case .certificateNotTrusted:
```
Comment (yim-lee) [inline Sources/PackageRegistry/SignatureValidation.swift:240]: Need this change: https://github.com/apple/swift-package-manager/pull/6237/files#diff-ac97ac612803fe11304731a730d285e4f7fd1cc5a2b139d3e91a50650833d58fR208

**Review by yim-lee — APPROVED** [review]

File: Sources/PackageRegistry/SignatureValidation.swift (hunk 1 above)
Comment (tomerd) [reply Sources/PackageRegistry/SignatureValidation.swift:240]: @yim-lee sorry I merged before notifying this, hope does not cause too much trouble

**Pull request merged**